test/datapipes/test_mesh_datapipe.py

```diff
@@ -0,0 +1,172 @@
+        triangle.GetPointIds().SetId(1, p2)
+        triangle.GetPointIds().SetId(2, p3)
+        triangles.InsertNextCell(triangle)
+
+    # Create a PolyData object (VTP mesh)
+    vtp_mesh = vtk.vtkPolyData()
+    vtp_mesh.SetPoints(points)
+    vtp_mesh.SetPolys(triangles)
+
+    # Assign random scalar values (features) to points in VTP mesh
+    scalar_values = vtk.vtkDoubleArray()
+    scalar_values.SetName("RandomFeatures")
+    for _ in range(num_points):
+        scalar_values.InsertNextValue(random.uniform(0, 1))
+    vtp_mesh.GetPointData().SetScalars(scalar_values)
+
+    # Write VTP mesh to file
+    vtp_writer = vtk.vtkXMLPolyDataWriter()
+    vtp_writer.SetFileName(os.path.join(dir, "random.vtp"))
+    vtp_writer.SetInputData(vtp_mesh)
+    vtp_writer.Write()
+
+    # Create an Unstructured Grid (VTU mesh)
+    vtu_mesh = vtk.vtkUnstructuredGrid()
+    vtu_mesh.SetPoints(points)
+    vtu_mesh.SetCells(vtk.VTK_TRIANGLE, triangles)
+
+    # Assign random scalar values (features) to points in VTU mesh
+    scalar_values = vtk.vtkDoubleArray()
+    scalar_values.SetName("RandomFeatures")  # Set the name
+    for _ in range(num_points):
+        scalar_values.InsertNextValue(random.uniform(0, 1))
+    vtu_mesh.GetPointData().SetScalars(scalar_values)
+
+    # Write VTU mesh to file
+    vtu_writer = vtk.vtkXMLUnstructuredGridWriter()
+    vtu_writer.SetFileName(os.path.join(dir, "random.vtu"))
+    vtu_writer.SetInputData(vtu_mesh)
+    vtu_writer.Write()
+
+
+@pytest.mark.parametrize("device", ["cuda", "cpu"])
+def test_mesh_datapipe(device, tmp_path):
+    """Tests the MeshDatapipe class with VTP and VTU files."""
+    tmp_dir = tmp_path / "temp_data"
+    tmp_dir.mkdir()
+    _create_random_vtp_vtu_mesh(num_points=20, num_triangles=40, dir=tmp_dir)
+    datapipe_vtp = MeshDatapipe(
+        data_dir=tmp_dir,
+        variables=["RandomFeatures"],
+        num_variables=1,
+        file_format="vtp",
+        stats_dir=None,
+        batch_size=1,
+        num_samples=1,
+        shuffle=True,
+        num_workers=1,
+        device=device,
+    )
+
+    assert len(datapipe_vtp) == 1
+    for data in datapipe_vtp:
+        assert data[0]["vertices"].shape == (1, 20, 3)
+        assert data[0]["x"].shape == (1, 20, 1)
+
+    datapipe_vtu = MeshDatapipe(
+        data_dir=tmp_dir,
+        variables=["RandomFeatures"],
+        num_variables=1,
+        file_format="vtu",
+        stats_dir=None,
+        batch_size=1,
+        num_samples=1,
+        shuffle=True,
+        num_workers=1,
+        device=device,
+    )
+
+    assert len(datapipe_vtu) == 1
+    for data in datapipe_vtu:
+        assert data[0]["vertices"].shape == (1, 20, 3)
+        assert data[0]["x"].shape == (1, 20, 1)
+
+
+@nfsdata_or_fail
+@pytest.mark.parametrize("device", ["cuda", "cpu"])
+def test_mesh_datapipe_cgns(device, pytestconfig):
+    """Tests the mesh datapipe for CGNS file format."""
+    datapipe_cgns = MeshDatapipe(
+        data_dir=cgns_data_dir,
```

Comment:
I think you forgot to add `cgns_data_dir` to the test function arguments? That's why the CI failed since `cgns_data_dir` is a fixture (function)?